cantools/database/can/c_source.py

```diff
@@ -1350,11 +1363,18 @@ def _generate_declarations(database_name, messages, floating_point_numbers):
             signal_declaration = ''
 
             if floating_point_numbers:
-                signal_declaration = SIGNAL_DECLARATION_ENCODE_DECODE_FMT.format(
-                    database_name=database_name,
-                    message_name=message.snake_name,
-                    signal_name=signal.snake_name,
-                    type_name=signal.type_name)
+                if not sender_name or sender_name in message.senders:
```

Comment:
I suggest adding a helper function `is_sender(message, node_name)` for less duplicated code and improved readability.

Reply:
Added with underscore as I think it shuold be internal:
_is_sender(message, node_name)